### ISSUE TITLE
grad_absmax=-inf

### ISSUE THREAD

**Issue by godwenbin** [bghira/SimpleTuner#1418]
2025-04-18 15:33:43,106 [INFO] Precomputing the negative prompt embed for validations.
2025-04-18 15:33:43,382 [INFO] Completed validation prompt gathering.
2025-04-18 15:33:43,384 [INFO] Calculated our maximum training steps at 2808 because we have 2 epochs and 1404 steps per epoch.
2025-04-18 15:33:43,384 [INFO] Collected the following data backends: ['text embeds.', '1224_dataset_flux']
2025-04-18 15:33:43,386 [INFO] Unloading text encoders, as they are not being trained.
2025-04-18 15:33:44,629 [INFO] After nuking text encoders from orbit, we freed 9.11 GB of VRAM.
2025-04-18 15:33:45,206 [INFO] After nuking the VAE from orbit, we freed 163.84 MB of VRAM.
2025-04-18 15:33:45,206 [INFO] Loading diffusion model from /mnt/nj-dev-data-image-text2image-sdb/project/flux1-schnell/model/FLUX.1-schnell
Loading checkpoint shards:   0%|                                                                     Loading checkpoint shards:   0%|                                                                     Loading checkpoint shards: 100%|███████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 37.83it/s]
Loading checkpoint shards: 100%|███████████████████████████████████████████████████████████████| 3/3 [00:00<00:00, 37.67it/s]
2025-04-18 15:33:45,537 [INFO] Moving the <class 'helpers.models.flux.transformer.FluxTransformer2DModelWithMasking'> to GPU in torch.bfloat16 precision.
2025-04-18 15:34:39,919 [INFO] Benchmarking base model for comparison. Supply `--disable_benchmark: true` to disable this behaviour.
2025-04-18 15:34:39,929 [INFO] Loading AutoencoderKL from /mnt/nj-dev-data-image-text2image-sdb/project/flux1-schnell/model/FLUX.1-schnell
Doing nothing, deepspeed zero3 was not enabled?
2025-04-18 15:34:40,087 [INFO] Moving AutoencoderKL to accelerator, converting from torch.float32 to torch.bfloat16

Processing validation prompts:   0%|                                                                 Processing validation prompts:   5%|██▊                                                       | 1/21 Processing validation prompts:  10%|█████▌                                                    | 2/21 Processing validation prompts:  14%|████████▎                                                 | 3/21 Processing validation prompts:  19%|███████████                                               | 4/21 Processing validation prompts:  24%|█████████████▊                                            | 5/21 Processing validation prompts:  29%|████████████████▌                                         | 6/21 Processing validation prompts:  33%|███████████████████▎                                      | 7/21 Processing validation prompts:  38%|██████████████████████                                    | 8/21 Processing validation prompts:  43%|████████████████████████▊                                 | 9/21 Processing validation prompts:  48%|███████████████████████████▏                             | 10/21 Processing validation prompts:  52%|█████████████████████████████▊                           | 11/21 Processing validation prompts:  57%|████████████████████████████████▌                        | 12/21 Processing validation prompts:  62%|███████████████████████████████████▎                     | 13/21 Processing validation prompts:  67%|██████████████████████████████████████                   | 14/21 Processing validation prompts:  71%|████████████████████████████████████████▋                | 15/21 Processing validation prompts:  76%|███████████████████████████████████████████▍             | 16/21 Processing validation prompts:  81%|██████████████████████████████████████████████▏          | 17/21 Processing validation prompts:  86%|████████████████████████████████████████████████▊        | 18/21 Processing validation prompts:  90%|███████████████████████████████████████████████████▌     | 19/21 Processing validation prompts:  95%|██████████████████████████████████████████████████████▎  | 20/21 Processing validation prompts: 100%|█████████████████████████████████████████████████████████| 21/21                         2025-04-18 15:40:37,291 [INFO] Evaluation result: {'clip/min': 16.232, 'clip/max': 28.8174, 'clip/mean': 24.162009523809527, 'clip/std': 3.047698410730044}
2025-04-18 15:40:47,742 [INFO] Learning rate: 0.0001
2025-04-18 15:40:47,742 [INFO] cls: <class 'accelerate.utils.deepspeed.DummyOptim'>, settings: {'lr': 0.0001, 'betas': (0.9, 0.999), 'eps': 1e-08, 'weight_decay': 0.01}
2025-04-18 15:40:47,859 [INFO] DeepSpeed Optimizer arguments, weight_decay=0.01 eps=1e-08, extra_arguments={'lr': 0.0001, 'betas': (0.9, 0.999), 'eps': 1e-08, 'weight_decay': 0.01}
2025-04-18 15:40:47,859 [INFO] Optimiser cannot use an LR scheduler, so we are disabling it.
2025-04-18 15:40:47,860 [INFO] Using dummy learning rate scheduler
2025-04-18 15:40:47,909 [INFO] Preparing models..
2025-04-18 15:40:47,910 [INFO] Loading our accelerator...
Using /root/.cache/torch_extensions/py310_cu121 as PyTorch extensions root...
Using /root/.cache/torch_extensions/py310_cu121 as PyTorch extensions root...
Emitting ninja build file /root/.cache/torch_extensions/py310_cu121/cpu_adam/build.ninja...
Building extension module cpu_adam...
Allowing ninja to set a default number of workers... (overridable by setting the environment variable MAX_JOBS=N)
ninja: no work to do.
Loading extension module cpu_adam...
Time to load cpu_adam op: 8.555764198303223 seconds
Adam Optimizer #0 is created with AVX512 arithmetic capability.
Config: alpha=0.000100, betas=(0.900000, 0.999000), weight_decay=0.010000, adam_w=1
Loading extension module cpu_adam...
Time to load cpu_adam op: 8.640222072601318 seconds
Adam Optimizer #0 is created with AVX512 arithmetic capability.
Config: alpha=0.000100, betas=(0.900000, 0.999000), weight_decay=0.010000, adam_w=1
2025-04-18 15:42:01,253 [INFO] After removing any undesired samples and updating cache entries, we have settled on 2 epochs and 1404 steps per epoch.
2025-04-18 15:42:06,183 [INFO] 
***** Running training *****
-  Num batches = 1404
-  Num Epochs = 2
  - Current Epoch = 1
-  Total train batch size (w. parallel, distributed & accumulation) = 2
  - Instantaneous batch size per device = 1
  - Gradient Accumulation steps = 1
-  Total optimization steps = 2808
-  Total optimization steps remaining = 2808
Epoch 1/2 Steps:   0%|                                                                              |Epoch 1/2, Steps:   0%|                                                                             |Epoch 1/2, Steps:   0%|                                                                  | 1/2808 [00Epoch 1/2, Steps:   0%|                          | 1/2808 [00:21<15:50:32, 20.32s/it, grad_absmax=-inEpoch 1/2, Steps:   0%|                          | 1/2808 [00:21<15:50:32, 20.32s/it, grad_absmax=-inEpoch 1/2, Steps:   0%|                          | 2/2808 [00:32<11:53:44, 15.26s/it, grad_absmax=-inEpoch 1/2, Steps:   0%|                          | 2/2808 [00:32<11:53:44, 15.26s/it, grad_absmax=-inEpoch 1/2, Steps:   0%|                          | 2/2808 [00:32<11:53:44, 15.26s/it, grad_absmax=-inEpoch 1/2, Steps:   0%|                          | 3/2808 [00:43<10:22:02, 13.31s/it, grad_absmax=-inEpoch 1/2, Steps:   0%|                          | 3/2808 [00:43<10:22:02, 13.31s/it, grad_absmax=-inEpoch 1/2, Steps:   0%|                          | 3/2808 [00:43<10:22:02, 13.31s/it, grad_absmax=-inEpoch 1/2, Steps:   0%|                           | 4/2808 [00:54<9:40:46, 12.43s/it, grad_absmax=-inEpoch 1/2, Steps:   0%|                          | 4/2808 [00:54<9:40:46, 12.43s/it, grad_absmax=-infEpoch 1/2, Steps:   0%|                          | 4/2808 [00:54<9:40:46, 12.43s/it, grad_absmax=-infEpoch 1/2, Steps:   0%|                          | 5/2808 [01:04<9:07:38, 11.72s/it, grad_absmax=-infEpoch 1/2, Steps:   0%|                           | 5/2808 [01:04<9:07:38, 11.72s/it, grad_absmax=-inEpoch 1/2, Steps:   0%|                           | 5/2808 [01:05<9:07:38, 11.72s/it, grad_absmax=-inEpoch 1/2, Steps:   0%|                           | 6/2808 [01:15<8:47:53, 11.30s/it, grad_absmax=-inEpoch 1/2, Steps:   0%|                           | 6/2808 [01:15<8:47:53, 11.30s/it, grad_absmax=-inEpoch 1/2, Steps:   0%|                           | 6/2808 [01:15<8:47:53, 11.30s/it, grad_absmax=-inEpoch 1/2, Steps:   0%|                           | 7/2808 [01:26<8:50:08, 11.36s/it, grad_absmax=-inEpoch 1/2, Steps:   0%|                          | 7/2808 [01:27<8:50:08, 11.36s/it, grad_absmax=-infEpoch 1/2, Steps:   0%|                          | 7/2808 [01:27<8:50:08, 11.36s/it, grad_absmax=-infEpoch 1/2, Steps:   0%|                          | 8/2808 [01:38<8:58:44, 11.54s/it, grad_absmax=-infEpoch 1/2, Steps:   0%|                          | 8/2808 [01:38<8:58:44, 11.54s/it, grad_absmax=-infEpoch 1/2, Steps:   0%|                          | 8/2808 [01:39<8:58:44, 11.54s/it, grad_absmax=-infEpoch 1/2, Steps:   0%|   Epoch 1/2, Steps:   0%|▏                         | 14/2808 [02:43<8:26:07, 10.87s/it, grad_absmax=-inf, lr=0, step_loss=2.33]
******************************
i have a flux-deepspeed full-finetune duty, but when it runs, shows grad_max = -inf.but when i not use deepspeed,the grad is normal

**Comment by bghira** [comment]
why train schnell?

**Comment by godwenbin** [comment]
@bghira schnell has a permissive license so i wonder full finetune a good performance schnell-db to use, can this success in your point? Also, after i change the root model to flux-dev, it also has a -inf grad with deepspeed:

**Comment by bghira** [comment]
is not an issue.

**Comment by godwenbin** [comment]
@bghira so you mean -inf grad can also train success?

**Comment by bghira** [comment]
yeah, deepspeed's abstraction layer prevents us from tracking this, as it handles its own gradient clipping

**Comment by ozanciga** [comment]
@bghira i've encountered this as well and i think it makes sense to add a warning to indicate that this is expected. the progress bar is not consistent across different optimizers (e.g., optimi-adam you get non -inf grad) and the loss updates as expected too so it gets very confusing (especially considering when most finetunes have text lr = 0)

**Comment by bghira** [comment]
okay, i have this fix now included in next changes.
